### PR TITLE
feature(main): add system config to sealos

### PR DESCRIPTION
**Pull request by cuisongliu**

### CODE REVIEW

File: pkg/system/config.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+func (c *systemConfig) Get(key string) (string, error) {
+	return getValueOrDefault(key)
+}
+
+func (c *systemConfig) Set(key, value string) error {
+	return setValue(key, value)
+}
+
+func NewConfig() Config {
```
Comment (fengxsong) [inline pkg/system/config.go:43]: initialize a global config getter/setter so caller don't need to invoke function `NewConfig` repeatedly.

File: pkg/system/config.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	"github.com/containers/buildah"
+	"github.com/containers/storage/pkg/homedir"
+)
+
+type Config interface {
+	Get(string) (string, error)
+	Set(string, string) error
+}
+
+type systemConfig struct{}
```
Comment (fengxsong) [inline pkg/system/config.go:33]: consider named it to `envSystemConfig`?

File: pkg/system/config_cmd.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+package system
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+func NewConfigCmd() *cobra.Command {
```
[the remark this answers is not in the document]
Comment (cuisongliu) [reply pkg/system/config_cmd.go:25]: > do we need to create a `alpha` subcommand to contains all of those experimental subcommands?

Yes, but we need to do it independently with a pr.

File: pkg/system/config.go (hunk 1 above)
Comment (cuisongliu) [reply pkg/system/config.go:43]: done

File: pkg/system/config.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	RuntimeRootConfigKey   = "sealos_runtime_root"
+	DataRootConfigKey      = "sealos_data_root"
+	BuildahFormatConfigKey = "buildah_format"
+)
+
+func getValueOrDefault(key string) (string, error) {
+	for _, option := range configOptions {
+		if option.Key == key {
+			if option.OSEnv != "" {
+				if value := os.Getenv(option.OSEnv); value != "" {
```
[the remark this answers is not in the document]
Comment (cuisongliu) [reply pkg/system/config.go:94]: done

**Review by fengxsong — APPROVED** [review]

**Pull request merged**